### ISSUE TITLE
Outlaw classes with max-classes-per-file

### ISSUE THREAD

**Issue by JustinChristensen** [eslint/eslint#14217]
Apologies if there is already a rule specifically designed to do this, but I was hoping to outlaw the use of ES6 classes altogether with ESLint, and `max-classes-per-file` seemed like the likely option to do that with.

```
ESLint: 7.22.0

Error: .eslintrc.js:
	Configuration for rule "max-classes-per-file" is invalid:
	Value 0 should be >= 1.
```

I feel triggered, to say the least. ;)

**What rule do you want to change?**

`max-classes-per-file`

**Does this change cause the rule to produce more or fewer warnings?**

Well, I suppose it would give it the ability to produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Allow for a value of 0 to be used in the rule configuration.

**Please provide some example code that this change will affect:**

```js
        'max-classes-per-file': ['error', 0], // errors, see above
```

**What does the rule currently do for this code?**

Triggers a configuration validation error.

**What will the rule do after it's changed?**

Allow for the outlawing of ES6 classes.

**Are you willing to submit a pull request to implement this change?**

Yes.

**Comment by mdjermanovic** [comment]
Hi @JustinChristensen, thanks for the issue!

If you want to disallow the use of some syntax, you can use the [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) rule.

For classes, the selector would be `"ClassDeclaration, ClassExpression"`:

```js
/* eslint no-restricted-syntax: ["error", "ClassDeclaration, ClassExpression"] */

class A {}; // error

const B = class {}; // error
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCBcIkNsYXNzRGVjbGFyYXRpb24sIENsYXNzRXhwcmVzc2lvblwiXSAqL1xuXG5jbGFzcyBBIHt9OyAvLyBlcnJvclxuXG5jb25zdCBCID0gY2xhc3Mge307IC8vIGVycm9yIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

It isn't likely that we'll change the `max-classes-per-file` rule to accept `0`, as it isn't what this rule was designed for.

**Comment by JustinChristensen** [comment]
@mdjermanovic Excellent. Thanks.
